airflow/models/dag.py

```diff
@@ -3505,3 +3556,57 @@ def get_current_dag(cls) -> DAG | None:
             return cls._context_managed_dags[0]
         except IndexError:
             return None
+
+
+def _run_task(ti: TaskInstance, session):
+    """
+    Run a single task instance, and push result to Xcom for downstream tasks. Bypasses a lot of
+    extra steps used in `task.run` to keep our local running as fast as possible
+    Args:
+        ti: TaskInstance to run
+    """
+    log.info("*****************************************************")
+    if ti.map_index > 0:
+        log.info("Running task %s index %d", ti.task_id, ti.map_index)
+    else:
+        log.info("Running task %s", ti.task_id)
+    try:
+        ti._run_raw_task(session=session)
+        session.flush()
+        log.info("%s ran successfully!", ti.task_id)
+    except AirflowSkipException:
+        log.info("Task Skipped, continuing")
+    log.info("*****************************************************")
+    ti.set_state(State.SUCCESS)
```

Comment:
i believe that `_run_raw_task` already catches skip exception....
... i guess... what do we do about that?
if it was running in the old way, as a backfill job, it would mark the task as skipped, and its downstreams would [probably]  be skipped [depending on trigger rules].  do you know what happens here?

Reply:
@dstandish I think that's what happens here. I can test it on some DAGs I know have skips soon

Reply:
can confirm: skips work

```
/Users/danielimberman/code/astro-sdk/python-sdk/example_dags/advanced.py:12 DeprecationWarning: The `airflow.operators.dummy.DummyOperator` class is deprecated. Please use `'airflow.operators.empty.EmptyOperator'`.
/Users/danielimberman/code/airflow/airflow/models/dag.py:3449 RemovedInAirflow3Warning: Param `schedule_interval` is deprecated and will be removed in a future release. Please use `schedule` instead.
/Users/danielimberman/code/astro-sdk/python-sdk/example_dags/advanced.py:144 RemovedInAirflow3Warning: Parameter ``use_task_execution_day`` is deprecated. Use ``use_task_logical_date``.
[2022-09-15 14:42:42,495] {dag.py:3596} INFO - dagrun id:example_dag_advanced
/Users/danielimberman/code/airflow/airflow/models/dag.py:3605 RemovedInAirflow3Warning: Calling `DAG.create_dagrun()` without an explicit data interval is deprecated
created dagrun <DagRun example_dag_advanced @ 2022-09-15T21:42:42.393239+00:00: manual__2022-09-15T21:42:42.393239+00:00, state:running, queued_at: None. externally triggered: False>
[2022-09-15 14:42:42,505] {dag.py:2483} INFO - starting dagrun
[2022-09-15 14:42:42,510] {dag.py:3573} INFO - *****************************************************
[2022-09-15 14:42:42,510] {dag.py:3577} INFO - Running task begin
/Users/danielimberman/code/airflow/airflow/models/taskinstance.py:873 SAWarning: TypeDecorator ExecutorConfigType() will not produce a cache key because the ``cache_ok`` attribute is not set to True.  This can have significant performance implications including some performance degradations in comparison to prior SQLAlchemy versions.  Set this attribute to True if this type object's state is safe to use in a cache key, or False to disable this warning. (Background on this error at: https://sqlalche.me/e/14/cprf)
[2022-09-15 14:42:42,643] {taskinstance.py:1590} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_OWNER=community
AIRFLOW_CTX_DAG_ID=example_dag_advanced
AIRFLOW_CTX_TASK_ID=begin
AIRFLOW_CTX_EXECUTION_DATE=2022-09-15T21:42:42.393239+00:00
AIRFLOW_CTX_TRY_NUMBER=1
AIRFLOW_CTX_DAG_RUN_ID=manual__2022-09-15T21:42:42.393239+00:00
/Users/danielimberman/code/airflow/airflow/models/taskinstance.py:871 SAWarning: TypeDecorator ExecutorConfigType() will not produce a cache key because the ``cache_ok`` attribute is not set to True.  This can have significant performance implications including some performance degradations in comparison to prior SQLAlchemy versions.  Set this attribute to True if this type object's state is safe to use in a cache key, or False to disable this warning. (Background on this error at: https://sqlalche.me/e/14/cprf)
[2022-09-15 14:42:42,651] {taskinstance.py:1401} INFO - Marking task as SUCCESS. dag_id=example_dag_advanced, task_id=begin, execution_date=20220915T214242, start_date=, end_date=20220915T214242
[2022-09-15 14:42:42,656] {dag.py:3581} INFO - begin ran successfully!
[2022-09-15 14:42:42,656] {dag.py:3584} INFO - *****************************************************
[2022-09-15 14:42:42,659] {dag.py:3573} INFO - *****************************************************
[2022-09-15 14:42:42,659] {dag.py:3577} INFO - Running task check_day_of_week
/Users/danielimberman/code/airflow/airflow/models/taskinstance.py:873 SAWarning: TypeDecorator ExecutorConfigType() will not produce a cache key because the ``cache_ok`` attribute is not set to True.  This can have significant performance implications including some performance degradations in comparison to prior SQLAlchemy versions.  Set this attribute to True if this type object's state is safe to use in a cache key, or False to disable this warning. (Background on this error at: https://sqlalche.me/e/14/cprf)
[2022-09-15 14:42:42,671] {taskinstance.py:1590} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_OWNER=community
AIRFLOW_CTX_DAG_ID=example_dag_advanced
AIRFLOW_CTX_TASK_ID=check_day_of_week
AIRFLOW_CTX_EXECUTION_DATE=2022-09-15T21:42:42.393239+00:00
AIRFLOW_CTX_TRY_NUMBER=1
AIRFLOW_CTX_DAG_RUN_ID=manual__2022-09-15T21:42:42.393239+00:00
[2022-09-15 14:42:42,671] {skipmixin.py:155} INFO - Following branch weekday
[2022-09-15 14:42:42,671] {skipmixin.py:190} INFO - Skipping tasks ['weekend']
/Users/danielimberman/code/airflow/airflow/models/taskinstance.py:871 SAWarning: TypeDecorator ExecutorConfigType() will not produce a cache key because the ``cache_ok`` attribute is not set to True.  This can have significant performance implications including some performance degradations in comparison to prior SQLAlchemy versions.  Set this attribute to True if this type object's state is safe to use in a cache key, or False to disable this warning. (Background on this error at: https://sqlalche.me/e/14/cprf)
[2022-09-15 14:42:42,682] {taskinstance.py:1401} INFO - Marking task as SUCCESS. dag_id=example_dag_advanced, task_id=check_day_of_week, execution_date=20220915T214242, start_date=, end_date=20220915T214242
[2022-09-15 14:42:42,684] {dag.py:3581} INFO - check_day_of_week ran successfully!
[2022-09-15 14:42:42,684] {dag.py:3584} INFO - *****************************************************
[2022-09-15 14:42:42,703] {dag.py:3573} INFO - *****************************************************
[2022-09-15 14:42:42,703] {dag.py:3577} INFO - Running task weekday
/Users/danielimberman/code/airflow/airflow/models/taskinstance.py:873 SAWarning: TypeDecorator ExecutorConfigType() will not produce a cache key because the ``cache_ok`` attribute is not set to True.  This can have significant performance implications including some performance degradations in comparison to prior SQLAlchemy versions.  Set this attribute to True if this type object's state is safe to use in a cache key, or False to disable this warning. (Background on this error at: https://sqlalche.me/e/14/cprf)
[2022-09-15 14:42:42,726] {taskinstance.py:1590} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_OWNER=community
AIRFLOW_CTX_DAG_ID=example_dag_advanced
AIRFLOW_CTX_TASK_ID=weekday
AIRFLOW_CTX_EXECUTION_DATE=2022-09-15T21:42:42.393239+00:00
AIRFLOW_CTX_TRY_NUMBER=1
AIRFLOW_CTX_DAG_RUN_ID=manual__2022-09-15T21:42:42.393239+00:00
/Users/danielimberman/code/airflow/airflow/models/taskinstance.py:871 SAWarning: TypeDecorator ExecutorConfigType() will not produce a cache key because the ``cache_ok`` attribute is not set to True.  This can have significant performance implications including some performance degradations in comparison to prior SQLAlchemy versions.  Set this attribute to True if this type object's state is safe to use in a cache key, or False to disable this warning. (Background on this error at: https://sqlalche.me/e/14/cprf)
[2022-09-15 14:42:42,728] {taskinstance.py:1401} INFO - Marking task as SUCCESS. dag_id=example_dag_advanced, task_id=weekday, execution_date=20220915T214242, start_date=, end_date=20220915T214242
[2022-09-15 14:42:42,730] {dag.py:3581} INFO - weekday ran successfully!
[2022-09-15 14:42:42,730] {dag.py:3584} INFO - *****************************************************
[2022-09-15 14:42:42,734] {dag.py:3573} INFO - *****************************************************
[2022-09-15 14:42:42,734] {dag.py:3577} INFO - Running task weekday_activities.which_weekday_activity_day
/Users/danielimberman/code/airflow/airflow/models/taskinstance.py:873 SAWarning: TypeDecorator ExecutorConfigType() will not produce a cache key because the ``cache_ok`` attribute is not set to True.  This can have significant performance implications including some performance degradations in comparison to prior SQLAlchemy versions.  Set this attribute to True if this type object's state is safe to use in a cache key, or False to disable this warning. (Background on this error at: https://sqlalche.me/e/14/cprf)
[2022-09-15 14:42:42,746] {taskinstance.py:1590} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_OWNER=community
AIRFLOW_CTX_DAG_ID=example_dag_advanced
AIRFLOW_CTX_TASK_ID=weekday_activities.which_weekday_activity_day
AIRFLOW_CTX_EXECUTION_DATE=2022-09-15T21:42:42.393239+00:00
AIRFLOW_CTX_TRY_NUMBER=1
AIRFLOW_CTX_DAG_RUN_ID=manual__2022-09-15T21:42:42.393239+00:00
[2022-09-15 14:42:42,746] {python.py:177} INFO - Done. Returned value was: weekday_activities.contributing_to_Airflow
[2022-09-15 14:42:42,746] {skipmixin.py:155} INFO - Following branch weekday_activities.contributing_to_Airflow
[2022-09-15 14:42:42,746] {skipmixin.py:190} INFO - Skipping tasks ['weekday_activities.guitar_lessons', 'weekday_activities.studying', 'weekday_activities.family_dinner', 'weekday_activities.soccer_practice']
/Users/danielimberman/code/airflow/airflow/models/taskinstance.py:871 SAWarning: TypeDecorator ExecutorConfigType() will not produce a cache key because the ``cache_ok`` attribute is not set to True.  This can have significant performance implications including some performance degradations in comparison to prior SQLAlchemy versions.  Set this attribute to True if this type object's state is safe to use in a cache key, or False to disable this warning. (Background on this error at: https://sqlalche.me/e/14/cprf)
[2022-09-15 14:42:42,758] {taskinstance.py:1401} INFO - Marking task as SUCCESS. dag_id=example_dag_advanced, task_id=weekday_activities.which_weekday_activity_day, execution_date=20220915T214242, start_date=, end_date=20220915T214242
[2022-09-15 14:42:42,761] {dag.py:3581} INFO - weekday_activities.which_weekday_activity_day ran successfully!
[2022-09-15 14:42:42,761] {dag.py:3584} INFO - *****************************************************
[2022-09-15 14:42:42,765] {dag.py:3573} INFO - *****************************************************
[2022-09-15 14:42:42,765] {dag.py:3577} INFO - Running task weekday_activities.contributing_to_Airflow
/Users/danielimberman/code/airflow/airflow/models/taskinstance.py:873 SAWarning: TypeDecorator ExecutorConfigType() will not produce a cache key because the ``cache_ok`` attribute is not set to True.  This can have significant performance implications including some performance degradations in comparison to prior SQLAlchemy versions.  Set this attribute to True if this type object's state is safe to use in a cache key, or False to disable this warning. (Background on this error at: https://sqlalche.me/e/14/cprf)
[2022-09-15 14:42:42,776] {taskinstance.py:1590} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_OWNER=community
AIRFLOW_CTX_DAG_ID=example_dag_advanced
AIRFLOW_CTX_TASK_ID=weekday_activities.contributing_to_Airflow
AIRFLOW_CTX_EXECUTION_DATE=2022-09-15T21:42:42.393239+00:00
AIRFLOW_CTX_TRY_NUMBER=1
AIRFLOW_CTX_DAG_RUN_ID=manual__2022-09-15T21:42:42.393239+00:00
[2022-09-15 14:42:42,777] {subprocess.py:63} INFO - Tmp dir root location:
 /var/folders/gz/nxcpfdvn03x1v_ww1ym52xq80000gn/T
[2022-09-15 14:42:42,778] {subprocess.py:75} INFO - Running command: ['/bin/bash', '-c', "echo It's Thursday and I'm busy with contributing to Airflow."]
[2022-09-15 14:42:42,796] {subprocess.py:86} INFO - Output:
[2022-09-15 14:42:42,810] {subprocess.py:93} INFO - Its Thursday and Im busy with contributing to Airflow.
[2022-09-15 14:42:42,810] {subprocess.py:97} INFO - Command exited with return code 0
/Users/danielimberman/code/airflow/airflow/models/taskinstance.py:871 SAWarning: TypeDecorator ExecutorConfigType() will not produce a cache key because the ``cache_ok`` attribute is not set to True.  This can have significant performance implications including some performance degradations in comparison to prior SQLAlchemy versions.  Set this attribute to True if this type object's state is safe to use in a cache key, or False to disable this warning. (Background on this error at: https://sqlalche.me/e/14/cprf)
[2022-09-15 14:42:42,817] {taskinstance.py:1401} INFO - Marking task as SUCCESS. dag_id=example_dag_advanced, task_id=weekday_activities.contributing_to_Airflow, execution_date=20220915T214242, start_date=, end_date=20220915T214242
[2022-09-15 14:42:42,819] {dag.py:3581} INFO - weekday_activities.contributing_to_Airflow ran successfully!
[2022-09-15 14:42:42,819] {dag.py:3584} INFO - *****************************************************
[2022-09-15 14:42:42,822] {dag.py:3573} INFO - *****************************************************
[2022-09-15 14:42:42,822] {dag.py:3577} INFO - Running task end
/Users/danielimberman/code/airflow/airflow/models/taskinstance.py:873 SAWarning: TypeDecorator ExecutorConfigType() will not produce a cache key because the ``cache_ok`` attribute is not set to True.  This can have significant performance implications including some performance degradations in comparison to prior SQLAlchemy versions.  Set this attribute to True if this type object's state is safe to use in a cache key, or False to disable this warning. (Background on this error at: https://sqlalche.me/e/14/cprf)
[2022-09-15 14:42:42,833] {taskinstance.py:1590} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_OWNER=community
AIRFLOW_CTX_DAG_ID=example_dag_advanced
AIRFLOW_CTX_TASK_ID=end
AIRFLOW_CTX_EXECUTION_DATE=2022-09-15T21:42:42.393239+00:00
AIRFLOW_CTX_TRY_NUMBER=1
AIRFLOW_CTX_DAG_RUN_ID=manual__2022-09-15T21:42:42.393239+00:00
/Users/danielimberman/code/airflow/airflow/models/taskinstance.py:871 SAWarning: TypeDecorator ExecutorConfigType() will not produce a cache key because the ``cache_ok`` attribute is not set to True.  This can have significant performance implications including some performance degradations in comparison to prior SQLAlchemy versions.  Set this attribute to True if this type object's state is safe to use in a cache key, or False to disable this warning. (Background on this error at: https://sqlalche.me/e/14/cprf)
[2022-09-15 14:42:42,835] {taskinstance.py:1401} INFO - Marking task as SUCCESS. dag_id=example_dag_advanced, task_id=end, execution_date=20220915T214242, start_date=, end_date=20220915T214242
[2022-09-15 14:42:42,837] {dag.py:3581} INFO - end ran successfully!
[2022-09-15 14:42:42,837] {dag.py:3584} INFO - *****************************************************
[2022-09-15 14:42:42,839] {dagrun.py:597} INFO - Marking run <DagRun example_dag_advanced @ 2022-09-15T21:42:42.393239+00:00: manual__2022-09-15T21:42:42.393239+00:00, state:running, queued_at: None. externally triggered: False> successful
[2022-09-15 14:42:42,839] {dagrun.py:644} INFO - DagRun Finished: dag_id=example_dag_advanced, execution_date=2022-09-15T21:42:42.393239+00:00, run_id=manual__2022-09-15T21:42:42.393239+00:00, run_start_date=2022-09-15 21:42:42.393239+00:00, run_end_date=2022-09-15 21:42:42.839475+00:00, run_duration=0.446236, state=success, external_trigger=False, run_type=manual, data_interval_start=2022-09-15T21:42:42.393239+00:00, data_interval_end=2022-09-16T00:00:00+00:00, dag_hash=None
```

Reply:
super 👍